### PR TITLE
Web API V1 documentation

### PR DESCRIPTION
**Pull request by drmathias**
Includes XML documentation in the Swagger UI and documents all web API responses and when they occur.

Example:

image

### CODE REVIEW

**Comment by codingupastorm** [comment]
Sorry for the delay in looking at this @drmathias ! Having a look now.

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs [hunk 1]
```diff
@@ -83,8 +87,12 @@ public IActionResult DeploymentFlags()
         /// </summary>
         /// <returns>Json formatted <see cref="uint256"/> hash of the block at the consensus tip. Returns <see cref="IActionResult"/> formatted error if fails.</returns>
         /// <remarks>This is an API implementation of an RPC call.</remarks>
+        /// <response code="200">Returns the block hash</response>
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs]:
```suggestion
        /// <response code="200">Returns the hash of the block at the consensus tip.</response>
```

File: src/Stratis.Bitcoin.Features.Miner/Controllers/MiningController.cs [hunk 2]
```diff
@@ -58,8 +58,18 @@ public MiningController(IConsensusManager consensusManager, IFullNode fullNode,
         /// <returns>List of block header hashes of newly mined blocks.</returns>
         /// <remarks>It is possible that less than the required number of blocks will be mined because the generating function only
         /// tries all possible header nonces values.</remarks>
+        /// <response code="200">Returns the generated block hashes</response>
+        /// <response code="400">Invalid request, or unexpected exception occured</response>
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Miner/Controllers/MiningController.cs]: `occurred`, the same typo is present in other places in the added documentation

File: src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs [hunk 3]
```diff
@@ -35,15 +35,19 @@ public NotificationsController(IBlockNotification blockNotification, ChainIndexe
         /// <returns>Http OK if the request succeeded.</returns>
         /// <example>/api/notifications/sync?from=1155695</example>
         /// <example>/api/notifications/sync?from=000000002f7105530e69b80f9295d3bc3046cd7d8643a2fb3aaaa23f90c82c77</example>
+        /// <response code="200">Syncronisation started</response>
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs]:
```suggestion
        /// <response code="200">Synchronisation started</response>
```

File: src/Stratis.Bitcoin/Controllers/NodeController.cs [hunk 4]
```diff
@@ -71,7 +71,7 @@ public class NodeController : Controller
         private readonly IGetUnspentTransaction getUnspentTransaction;
 
         /// <summary>Specification of the network the node runs on.</summary>
-        private Network network; // Not readonly because of ValidateAddress
+        private readonly Network network; // Not readonly because of ValidateAddress
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Controllers/NodeController.cs]: Might as well remove the comment from here, I can't see what it was referring to in the first place
```suggestion
        private readonly Network network;
```

**Pull request merged**